### ISSUE TITLE
Trying to build the conda-forge recipe for conda with conda-build 3 self-implodes the root environment

### ISSUE THREAD

**Issue by ericdill** [conda/conda-build#2206]
recipe: https://github.com/conda-forge/conda-feedstock

End result is that conda-build fails to build the recipe because, somehow, conda becomes this:

```
$ cat `which conda`
#!/home/p-edill/miniconda/conda-bld/conda_1500217707105/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/python

# -*- coding: utf-8 -*-
import re
import sys

from conda.cli import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

And the error from conda build
```
++ args=/home/p-edill/miniconda/conda-bld/conda_1500217707105/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p
++ /home/p-edill/miniconda/bin/conda ..checkenv bash /home/p-edill/miniconda/conda-bld/conda_1500217707105/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p
/home/p-edill/miniconda/bin/activate: /home/p-edill/miniconda/bin/conda: /home/p-edill/miniconda/conda-bld/conda_1500217707105/_h_env_placehold_placeh: bad interpreter: No such file or directory
TESTS FAILED: conda-4.3.22-py36hb41f66f_0.tar.bz2
```

conda version 4.3.22
conda build version 3.0.6

**Comment by msarahan** [comment]
This is pretty confusing.  The conda entry point script actually doesn't look wrong.  The tricky part is that there's a mismatch in the prefix length - probably some error with a package that didn't have a long prefix?  Let's take a look at your logs here when you have time.

**Comment by msarahan** [comment]
fixed by #2209

**Comment by ericdill** [comment]
Thanks @msarahan !
